code/config/lib/Loot/Config/Yaml.hs

```diff
@@ -95,3 +98,71 @@ parseMulti p = keyOrDefault (fromString $ symbolVal p) mempty configParser
 
 instance OptionsFromJson is => FromJSON (ConfigRec 'Partial is) where
     parseJSON = toAesonParser' configParser
+
+class OptionsToJsonList (k :: ConfigKind) (is :: [ItemKind]) where
+  configToJsonList :: ConfigRec k is -> [A.Pair]
+
+instance OptionsToJsonList (k :: ConfigKind) '[] where
+  configToJsonList RNil = []
+
+undefinedValue :: A.Value
+undefinedValue = toJSON $ fromString @Text "<undefined>"
+
+instance
+  forall l v is k.
+        ( KnownSymbol l
+        , ToJSON v
+        , OptionsToJsonList k is)
+    => OptionsToJsonList k ((l ::: v) ': is)
+  where
+    configToJsonList (itemOption :& rest) =
+      let value = case itemOption of
+            ItemOptionP mv -> maybe undefinedValue toJSON mv
+            ItemOptionF v -> toJSON v
+      in (fromString $ symbolVal (Proxy :: Proxy l), value) : configToJsonList @k @is rest
+
+instance
+    forall l us is k.
+        ( KnownSymbol l
+        , OptionsToJsonList k us
+        , OptionsToJsonList k is
+        )
+    => OptionsToJsonList k ((l ::< us) ': is)
+  where
+    configToJsonList (ItemSub inner :& rest) =
+      ( fromString $ symbolVal (Proxy :: Proxy l)
+      , A.object $ configToJsonList inner
+      ) : configToJsonList rest
+
+instance
+    forall l us is k.
+        ( KnownSymbol l
+        , OptionsToJsonList k  (SumSelection l : us)
+        , OptionsToJsonList k is
+        )
+    => OptionsToJsonList k ((l ::+ us) ': is)
+  where
+    configToJsonList (itemSum :& rest) =
+      let value = case itemSum of
+            ItemSumP inner -> A.object $ configToJsonList inner
+            ItemSumF inner -> A.object $ configToJsonList inner
```

Comment:
This might be implemented semantically wrong, because I haven't understood meaning of this

Reply:
I have no idea either 😅. I believe, @pasqu4le added sum types.

Reply:
Yep, I added support for sum types, although quite some time ago, so I had to check the code again because I forgot how :sweat_smile: 

Anyway, this seems to be semantically correct.

If the confusion is about why there are 2 `ItemSum` I am not entirely sure, I recall something about them requiring different constraints but I cannot find this now and judging by some comments in the PR this difference might have been solved at some point.

In other terms, unless I am missing something, it looks like we could have had a single:
```haskell
    ItemSum       :: Item' k        (l ::+ is) -> Item k        (l ::+ is)
```
just like we have a single `ItemSub` and the implementation here is indeed correct.

P.s. I could check this for sure and open up another PR to fix this if necessary